test/integration/document_collection_test.rb

```diff
@@ -34,18 +24,58 @@ class DocumentCollectionTest < ActionDispatch::IntegrationTest
     assert page.has_text?("Each regime page provides a current list of asset freeze targets designated by the United Nations (UN), European Union and United Kingdom, under legislation relating to current financial sanctions regimes.")
   end
 
-  test "renders contents with link to each collection group" do
+  test "renders a contents list for each content item for every document collection group" do
     setup_and_visit_content_item("document_collection")
+    assert_equal 6, @content_item["details"]["collection_groups"].size
+
     @content_item["details"]["collection_groups"].each do |group|
       assert page.has_css?("nav a", text: group["title"])
     end
   end
 
-  test "renders a content list" do
+  test "renders a content list with 6 items, ignoring any empty groups" do
```

Comment:
Ignores document collection groups that have no documents when presenting the contents list.
```
setup_and_visit_content_item("document_collection")
@content_item["details"]["collection_groups"] << { "title" => "Empty Group", "documents" => [] }
assert_equal 7, @content_item["details"]["collection_groups"].size

content_list_items = all("nav.gem-c-contents-list .gem-c-contents-list__list-item")
assert_equal 6, content_list_items.size

@content_item["details"]["collection_groups"].each do |group|
      next if group["documents"].empty?

      assert page.has_css?("nav a", text: group["title"])
end
```